test/functional/p2p_add_connections.py

```diff
@@ -14,6 +14,13 @@ def check_node_connections(*, node, num_in, num_out):
     assert_equal(info["connections_in"], num_in)
     assert_equal(info["connections_out"], num_out)
 
+class P2PFeelerReceiver(P2PInterface):
+    def on_version(self, message):
+        # The bitcoind node closes feeler connections as soon as a version
+        # message is received from the test framework. Don't send any responses
+        # to the node's version message since the connection will already be
+        # closed.
+        pass
```

Comment:
Since the node will disconnect as soon as it receives our `version` message, there's a race between it disconnecting and the `P2PInterface` sending the responses to its `version` message here: https://github.com/bitcoin/bitcoin/blob/eb09c26724e3f714b613788fc506f2ff3a208d2c/test/functional/test_framework/p2p.py#L435-L441. If it disconnects first, those calls to `send_message()` may fail, causing the test to fail.